packages/opentelemetry-plugin-user-interaction/src/userInteraction.ts

```diff
@@ -0,0 +1,457 @@
+/*!
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as shimmer from 'shimmer';
+import { BasePlugin, hrTime, isWrapped } from '@opentelemetry/core';
+import * as types from '@opentelemetry/types';
+import { getElementXPath } from '@opentelemetry/web';
+import {
+  AsyncTask,
+  RunTaskFunction,
+  UIMem,
+  ZoneTypeWithPrototype,
+} from './types';
+import { AttributeNames } from './enums/AttributeNames';
+
+const ZONE_SCOPE_KEY = 'OT_ZONE_SCOPE';
+const EVENT_CLICK_NAME = 'event_click:';
+const EVENT_NAVIGATION_NAME = 'Navigation:';
+
+/**
+ * This class represents a UserInteraction plugin for auto instrumentation.
+ * If zone.js is available then it patches the zone otherwise it patches
+ * addEventListener of HTMLElement
+ */
+export class UserInteractionPlugin extends BasePlugin<unknown> {
+  readonly component: string = 'user-interaction';
+  // @TODO align this with all packages #600
+  readonly version: string = '0.3.1';
+  moduleName = this.component;
+  private _uiMem = new WeakMap<types.Span, UIMem>();
+  private _zonePatched = false;
+
+  /**
+   * This will check if last task was timeout and will save the time to
+   * fix the user interaction when nothing happens
+   * This timeout comes from xhr plugin which is needed to collect information
+   * about last xhr main request from observer
+   * @param task
+   * @param span
+   */
+  private _checkForTimeout(task: AsyncTask, span: types.Span) {
+    const uiMem = this._uiMem.get(span);
+    if (uiMem) {
+      if (task.source === 'setTimeout') {
+        uiMem.hrTimeLastTimeout = hrTime();
+      } else if (
+        task.source !== 'Promise.then' &&
+        task.source !== 'setTimeout'
+      ) {
+        uiMem.hrTimeLastTimeout = undefined;
+      }
+    }
+  }
+
+  /**
+   * Creates a new span
+   * @param element
+   * @param eventName
+   */
+  private _createSpan(
+    element: HTMLElement,
+    eventName: string
+  ): types.Span | undefined {
+    if (!element.getAttribute) {
+      return undefined;
+    }
+    if (element.hasAttribute('disabled')) {
```

Comment:
Do you think it would be useful to include an additional element attribute that this plugin will look for to ignore tracking?

Reply:
Hmm don't have really opinion on that. Is there already anything like "common practice" or some standard how certain elements is being excluded from auto instrumentation ?. Do you think it should be configurable in config ?